### PR TITLE
Methods section

### PR DESCRIPTION
**Pull request by WebstudioThomHubers**

### CODE REVIEW

**Comment by WebstudioThomHubers** [comment]
#10

File: Documents/idea document 2/ideadocument2.md [hunk 1]
```diff
@@ -5,9 +5,38 @@
...
+This chapter specifies which methods will be used for the neighbourhood classification. The several steps that have to be taken to extract useful information out of the data are dispayed in a pipeline diagram, after which the steps are elaborated somewhat more.
 
-# Speficications & Execution Plan
+### Overall system architecture
+The following graph shows how input data (e.g. tweets) are processed into neighbourhood classifications.
+<p align="center">
+  <img alt="Workbench electron-browser component" src="../images/pipeline.png" width="60%">
+</p>
+
+These tweets can be used for both the processing and identification of neighbourhoods and direct display on the user interface. All relevant (sport-related) tweets will be displayed.
```
Comment (rpjproost) [inline Documents/idea document 2/ideadocument2.md]: I think relevant needs to be determined by for example BM25 algorithm with certain threshold (to be set). Then it is less vague

**Review by rpjproost — CHANGES_REQUESTED** [review]

**Comment by rpjproost** [comment]
When merged Closes #10

File: Documents/idea document 2/ideadocument2.md [hunk 2]
```diff
@@ -5,9 +5,38 @@
...
 # Methods
-(Thom)
+This chapter specifies which methods will be used for the neighbourhood classification. The several steps that have to be taken to extract useful information out of the data are dispayed in a pipeline diagram, after which the steps are elaborated somewhat more.
 
-# Speficications & Execution Plan
+### Overall system architecture
+The following graph shows how input data (e.g. tweets) are processed into neighbourhood classifications.
+<p align="center">
+  <img alt="Workbench electron-browser component" src="../images/pipeline.png" width="60%">
+</p>
```
Comment (rubenstarmans) [inline Documents/idea document 2/ideadocument2.md]: Add figure title

File: Documents/idea document 2/ideadocument2.md [hunk 3]
```diff
@@ -5,9 +5,38 @@
...
+
+### Text processing
+Messages are matched and indexed to a sports-related taxonomy to indicate whether the message gives information about a sports activity and which activity that could be.
+
+### Regionalisation
+After the contents of the message have been processed this information is sent to the regionalisation subsystem. This system clusters the messages over multple dimensions (spatial, time, sport). The spatial aspect of this is done by the maxp regionalisation algorithm, which is part of the [pysal library](http://pysal.readthedocs.io/en/latest/library/region/maxp.html). Clustering on the other properties is done by a more common clustering algorhitm like [k-means](http://scikit-learn.org/stable/modules/generated/sklearn.cluster.k_means.html#sklearn.cluster.k_means).
+
+### User Interface
+The tweets and regions are sent to the user interface. The different stakeholders can all retrieve their information need from it.
+<p align="center">
```
Comment (rubenstarmans) [inline Documents/idea document 2/ideadocument2.md]: Mockup moest bij expected outcomes komen te staan toch. Misschien hier een algemener verhaal over hoe de UI zal worden gebouwd met lapbox.

File: Documents/idea document 2/ideadocument2.md (hunk 1 above)
Comment (WebstudioThomHubers) [reply Documents/idea document 2/ideadocument2.md]: Will do this

File: Documents/idea document 2/ideadocument2.md (hunk 3 above)
Comment (WebstudioThomHubers) [reply Documents/idea document 2/ideadocument2.md]: Oke dan haal ik hem hier weg, zet degene die expected outcomes doet hem daarbij?

File: Documents/idea document 2/ideadocument2.md (hunk 2 above)
Comment (WebstudioThomHubers) [reply Documents/idea document 2/ideadocument2.md]: Will do

File: Documents/idea document 2/ideadocument2.md (hunk 3 above)
Comment (rubenstarmans) [reply Documents/idea document 2/ideadocument2.md]: Yes, dat was ik. Daar staat hij al ;)

**Review by rubenstarmans — APPROVED** [review]

File: Documents/idea document 2/ideadocument2.md [hunk 4]
```diff
@@ -13,7 +13,33 @@ With these two sources data which is relevant for the application, can be collec
 To derive this information after the collection, data must be analyzed, which will be discussed in the Section Methods.
 
 # Methods
-(Thom)
+This chapter specifies which methods will be used for the neighbourhood classification. The several steps that have to be taken to extract useful information out of the data are dispayed in a pipeline diagram, after which the steps are elaborated somewhat more.
```
Comment (daanschipper) [inline Documents/idea document 2/ideadocument2.md]: chapter -> section

File: Documents/idea document 2/ideadocument2.md [hunk 5]
```diff
@@ -13,7 +13,33 @@ With these two sources data which is relevant for the application, can be collec
...
+
+### Overall system architecture
+The following graph shows how input data (e.g. tweets) are processed into neighbourhood classifications.
+<p align="center">
+  <img alt="Workbench electron-browser component" src="../images/pipeline.png" width="60%"><br/>
+  <i>Graphical representation of the proposed system architecture</i>   
+</p>
```
Comment (daanschipper) [inline Documents/idea document 2/ideadocument2.md]: Proper caption, and reference
_Figure figure-thom-pipeline: Graphical representation of the proposed system architecture_

Comment (daanschipper) [reply Documents/idea document 2/ideadocument2.md]: We've identified to use Twitter and Strava as the source right? Could you update that in the pipeline

File: Documents/idea document 2/ideadocument2.md [hunk 6]
```diff
@@ -13,7 +13,33 @@ With these two sources data which is relevant for the application, can be collec
...
+
+These tweets can be used for both the processing and identification of neighbourhoods and direct display on the user interface. All relevant (sport-related) tweets will be displayed.
```
Comment (daanschipper) [inline Documents/idea document 2/ideadocument2.md]: Not only tweets

File: Documents/idea document 2/ideadocument2.md [hunk 7]
```diff
@@ -13,7 +13,33 @@ With these two sources data which is relevant for the application, can be collec
...
+
+The tweets are processed to determine which topics are discussed. Relevant (i.e. matching with a sport related word list) tweets are used to classify neighbourhoods. The clusters are also used in the user interface.
```
Comment (daanschipper) [inline Documents/idea document 2/ideadocument2.md]: again not only tweets

File: Documents/idea document 2/ideadocument2.md [hunk 8]
```diff
@@ -13,7 +13,33 @@ With these two sources data which is relevant for the application, can be collec
...
+
+### Crawler
+The crawler uses the Twitter API and Strava data. After location based filtering all data is passed to the Database Manager which stores it for later usage.
```
Comment (daanschipper) [inline Documents/idea document 2/ideadocument2.md]: Twitter and Strava API

File: Documents/idea document 2/ideadocument2.md [hunk 9]
```diff
@@ -13,7 +13,33 @@ With these two sources data which is relevant for the application, can be collec
...
+
+### Database Manager
+The storage is provided by a simple PostgreSQL database.
```
Comment (daanschipper) [inline Documents/idea document 2/ideadocument2.md]: ~~simple~~

File: Documents/idea document 2/ideadocument2.md [hunk 10]
```diff
@@ -13,7 +13,33 @@ With these two sources data which is relevant for the application, can be collec
...
+
+### Text processing
+Messages are matched and indexed to a sports-related taxonomy to indicate whether the message gives information about a sports activity and which activity that could be, this is done using the [BM25 algorhitm or similar](http://scikit-learn.org/dev/modules/feature_extraction.html#text-feature-extraction).
```
Comment (daanschipper) [inline Documents/idea document 2/ideadocument2.md]: messages -> queries

Comment (daanschipper) [reply Documents/idea document 2/ideadocument2.md]: ~~, this is done~~

File: Documents/idea document 2/ideadocument2.md [hunk 11]
```diff
@@ -13,7 +13,33 @@ With these two sources data which is relevant for the application, can be collec
...
+
+### Regionalisation
+After the contents of the message have been processed this information is sent to the regionalisation subsystem. This system clusters the messages over multple dimensions (spatial, time, sport). The spatial aspect of this is done by the maxp regionalisation algorithm, which is part of the [pysal library](http://pysal.readthedocs.io/en/latest/library/region/maxp.html). Clustering on the other properties is done by a more common clustering algorhitm like [k-means](http://scikit-learn.org/stable/modules/generated/sklearn.cluster.k_means.html#sklearn.cluster.k_means).
```
Comment (daanschipper) [inline Documents/idea document 2/ideadocument2.md]: ~~messages~~

Comment (daanschipper) [reply Documents/idea document 2/ideadocument2.md]: have been processed,

Comment (daanschipper) [reply Documents/idea document 2/ideadocument2.md]: this information? which information

Comment (daanschipper) [reply Documents/idea document 2/ideadocument2.md]: subsystem. This -> subsystem, which

Comment (daanschipper) [reply Documents/idea document 2/ideadocument2.md]: "Clustering on the other properties is done by a more common clustering algorhitm like k-means" Not possible to use, data has to be uniform distributed

File: Documents/idea document 2/ideadocument2.md [hunk 12]
```diff
@@ -13,7 +13,33 @@ With these two sources data which is relevant for the application, can be collec
...
+
+### User Interface
+The tweets and regions are sent to the user interface. The different stakeholders can all retrieve their information need from it. [Mapbox](https://www.mapbox.com/) is used to display all this on a map. After the basic map of Amsterdam is loaded, the areas are plotted on it, including the dots which represent messages that are part of that cluster. A comprehensive interface provides methods for filtering the areas based on time and sport.
```
Comment (daanschipper) [inline Documents/idea document 2/ideadocument2.md]: ~~tweets~~

Comment (daanschipper) [reply Documents/idea document 2/ideadocument2.md]: their information need from it, phrasing. can retrieve information they are interested in (or something like that)

Comment (daanschipper) [reply Documents/idea document 2/ideadocument2.md]: ~~basic~~

**Review by daanschipper — CHANGES_REQUESTED** [review]

File: Documents/idea document 2/ideadocument2.md (hunk 5 above)
Comment (WebstudioThomHubers) [reply Documents/idea document 2/ideadocument2.md]: I have to do Figure X right? What is x?

File: Documents/idea document 2/ideadocument2.md (hunk 11 above)
Comment (WebstudioThomHubers) [reply Documents/idea document 2/ideadocument2.md]: Would this be OK?

Clustering on the other properties is done by a more common clustering algorhitm, of which examples are included in the [Python sklearn package](http://scikit-learn.org/stable/modules/classes.html#module-sklearn.cluster)

File: Documents/idea document 2/ideadocument2.md (hunk 5 above)
Comment (daanschipper) [reply Documents/idea document 2/ideadocument2.md]: That will be changed at the end, but it is easier to search for "figure-thom-pipeline" and replace all to "X" than scrolling through the text

File: Documents/idea document 2/ideadocument2.md (hunk 11 above)
Comment (daanschipper) [reply Documents/idea document 2/ideadocument2.md]: Yes!

**Review by rpjproost — APPROVED** [review]

**Review by daanschipper — APPROVED** [review]

**Pull request merged**